### ISSUE TITLE
Support compression transfer data

### ISSUE THREAD

**Issue by tacyuuhon** [Vertamedia/chproxy#16]
I use the [clickhouse-jdbc](https://github.com/yandex/clickhouse-jdbc/tree/release_0.1.34) to write the data to the Clickhouse(1.1.54318) 
I will set the properties compress and decompress as true. 
The jdbc driver uses LZ4 to compress and decompress the data. And it can transfer data with Clickhouse(1.1.54318).
When I connect to chproxy, it doesn't work properly.

> error log
```
Caused by: java.io.IOException: Magic is not correct: 103
	at ru.yandex.clickhouse.response.ClickHouseLZ4Stream.readNextBlock(ClickHouseLZ4Stream.java:93) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.response.ClickHouseLZ4Stream.checkNext(ClickHouseLZ4Stream.java:74) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.response.ClickHouseLZ4Stream.read(ClickHouseLZ4Stream.java:50) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.response.StreamSplitter.readFromStream(StreamSplitter.java:85) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.response.StreamSplitter.next(StreamSplitter.java:47) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.response.ClickHouseResultSet.<init>(ClickHouseResultSet.java:65) ~[clickhouse-jdbc-0.1.34.jar:na]
	at ru.yandex.clickhouse.ClickHouseStatementImpl.executeQuery(ClickHouseStatementImpl.java:117) ~[clickhouse-jdbc-0.1.34.jar:na]
```

> config.yml
```
hack_me_please: false

server:
  http:
      listen_addr: "0.0.0.0:9090"
      allowed_networks: ["192.168.1.0/24"]
  metrics:
      allowed_networks: ["192.168.1.0/24"]

users:
  - name: "default"
    to_cluster: "default"
    to_user: "default"
    allow_cors: true
```

**Comment by hagen1778** [comment]
Hi @tacyuuhon 
I think this is security issue - all request params, except of few, are deleted by proxy.
I will take a look on it

**Comment by tacyuuhon** [comment]
:+1: Thank you fix it.

**Comment by valyala** [comment]
@tacyuuhon , the feature has been merged into `master` branch and will be available in the next release.

**Comment by valyala** [comment]
@tacyuuhon , try the [release 1.10.0](https://github.com/Vertamedia/chproxy/releases/tag/1.10.0)

**Comment by tacyuuhon** [comment]
@valyala @hagen1778 
I'll try to use the new version. 
Merry Christmas.

**Comment by hagen1778** [comment]
thx!

**Comment by valyala** [comment]
@tacyuuhon , the `1.10.0` release couldn't cache responses for compressed queries with `decompress=1` set. This should be fixed in the [1.10.1 release](https://github.com/Vertamedia/chproxy/releases/tag/1.10.1).

**Comment by tacyuuhon** [comment]
:thumbsup: I will use [1.10.1 release](https://github.com/Vertamedia/chproxy/releases/tag/1.10.1)
